src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -2242,7 +2243,7 @@ void Schedule::invalidNamePattern( const std::string& namePattern,  std::size_t
         }
 
         for (const auto& gname : root_group.groups()) {
-            auto child_group = this->groupTree(gname, report_step, std::addressof(tree));
+            auto child_group = this->groupTree(gname, report_step, std::make_shared<GTNode>( tree ));
```

Comment:
What your change is doing is creating a copy of the node for each entry in `root_group.groups()`. Seems like this means that trees in the `child_groups` point to different objects. Is that intended?

Wouldn't it better to change to `std::shared_ptr<GTNode> tree(root_group, parent)' in line 2237 and use that?

Reply:
> Seems like this means that trees in the child_groups point to different objects.

Yes you are right.

> Is that intended?

No - that is not intended; it should not affect correctness - but it sure is a weird tree. 

> Wouldn't it better to change to `std::shared_ptr tree(root_group, parent)' in line 2237 and use that?

Maybe - I'll look into it; would like to not export the pointer semantics though.